src/DefinitionStorage.php

```diff
@@ -96,12 +96,38 @@ public function set(string $id, mixed $definition): void
      *
      * @throws CircularReferenceException
      */
-    private function isResolvable(string $id, array $building): bool
+    private function isResolvable(string $id, array $building, ?string $parameterName = null): bool
     {
         if (isset($this->definitions[$id])) {
             return true;
         }
 
+        if (
```

Comment:
I assume that feature can degrade performance. Suggest to do it optional.

Reply:
@xepozz would you mind running a comparison benchmark for it?